File: 2463-minimum-recolors-to-get-k-consecutive-black-blocks/minimum-recolors-to-get-k-consecutive-black-blocks.py
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        L, pro = 0, 0
        minPro = len(blocks)
        for R in range(len(blocks)):
            if blocks[R] == 'W':
                pro += 1 
            if R - L == k-1:
                print(pro, R)
                minPro = min(minPro, pro)
                if blocks[L] == 'W':
                    pro -= 1
                L += 1

        return minPro

File: 2463-minimum-recolors-to-get-k-consecutive-black-blocks/minimum-recolors-to-get-k-consecutive-black-blocks.cpp
class Solution {
public:
    int minimumRecolors(std::string blocks, int k) {
        int L = 0, numWhites = 0; 
        int minRecolors = blocks.size(); 

        for (int R = 0; R < blocks.size(); ++R) {
            if (blocks[R] == 'W') {
                numWhites++; 
            }

            
            if (R >= k - 1) {
                minRecolors = std::min(minRecolors, numWhites); 

                if (blocks[L] == 'W') {
                    numWhites--; 
                }
                L++; 
            }
        }

        return minRecolors; 
    }
};